openfl/federated/data/sources/s3_data_source.py

```
@@ -16,14 +14,17 @@ class S3DataSource(DataSource):
 
     def __init__(
         self,
+        name,
         uri: str,
         endpoint=None,
         access_key_env_name=None,
         secret_key_env_name=None,
         secret_name=None,
         hash_func=None,
     ):
-        super().__init__(DataSourceType.S3)
+        import boto3
```

As Teo suggested in another comment, let's raise an exception telling user the package they should install.